python/pyspark/pandas/supported_api_gen.py

```diff
@@ -0,0 +1,377 @@
+
+    Write supported APIs documentation.
+    """
+    if LooseVersion(pd.__version__) < LooseVersion("1.4.0"):
+        warnings.warn(
+            "Warning: Latest version of pandas(>=1.4.0) is required to generate the documentation; "
+            + "however, your version was %s" % pd.__version__,
+            UserWarning,
+        )
+
+    all_supported_status: Dict[Tuple[str, str], Dict[str, SupportedStatus]] = {}
+    for pd_module_group, ps_module_group in MODULE_GROUP_MATCH:
+        pd_modules = _get_pd_modules(pd_module_group)
+        _update_all_supported_status(
+            all_supported_status, pd_modules, pd_module_group, ps_module_group
+        )
+    _write_rst(output_rst_file_path, all_supported_status)
+
+
+def _create_supported_by_module(
+    module_name: str, pd_module_group: Any, ps_module_group: Any
+) -> Dict[str, SupportedStatus]:
+    """
+    Retrieves supported status of pandas module
+
+    Parameters
+    ----------
+    module_name : str
+        Class name that exists in the path of the module.
+    pd_module_group : Any
+        Specific path of importable pandas module.
+    ps_module_group: Any
+        Specific path of importable pyspark.pandas module.
+    """
+    pd_module = getattr(pd_module_group, module_name) if module_name else pd_module_group
+    try:
+        ps_module = getattr(ps_module_group, module_name) if module_name else ps_module_group
+    except AttributeError:
+        # module not implemented
+        return {}
+
+    pd_funcs = dict([m for m in getmembers(pd_module, isfunction) if not m[0].startswith("_")])
+    if not pd_funcs:
+        return {}
+
+    ps_funcs = dict([m for m in getmembers(ps_module, isfunction) if not m[0].startswith("_")])
+
+    return _organize_by_implementation_status(
+        module_name, pd_funcs, ps_funcs, pd_module_group, ps_module_group
+    )
+
+
+def _organize_by_implementation_status(
+    module_name: str,
+    pd_funcs: Dict[str, Callable],
+    ps_funcs: Dict[str, Callable],
+    pd_module_group: Any,
+    ps_module_group: Any,
+) -> Dict[str, SupportedStatus]:
+    """
+    Check the implementation status and parameters of both modules.
+
+    Parmeters
+    ---------
+    module_name : str
+        Class name that exists in the path of the module.
+    pd_funcs: Dict[str, Callable]
+        function name and function object mapping of pandas module.
+    ps_funcs: Dict[str, Callable]
+        function name and function object mapping of pyspark.pandas module.
+    pd_module_group : Any
+        Specific path of importable pandas module.
+    ps_module_group: Any
+        Specific path of importable pyspark.pandas module.
+    """
+    pd_dict = {}
+    for pd_func_name, pd_func in pd_funcs.items():
+        ps_func = ps_funcs.get(pd_func_name)
+        if ps_func:
+            missing_set = (
+                set(signature(pd_func).parameters)
+                - set(signature(ps_func).parameters)
+                - COMMON_PARAMETER_SET
```

Comment:
Not related and might need us do some consideration in future.

- This is a good enough PR to generate what the pandas have but PS doesn't have. But I think we also need a way to find what PS have but Pandas not have (or to say PS specific parameter): `set(signature(ps_func).parameters)` - `set(signature(pd_func).parameters)`. Such as for Series:

- A further step, because we are using `set` rather than `list` in here, so we might loose some sorted info. So we might want to add a parameter sort check for `Y` func to aovid some potential problems like
https://github.com/apache/spark/pull/36414#discussion_r863508964 in future. But I think it's not a common case.

Reply:
oh, the first point is commented in https://github.com/apache/spark/pull/36509#issuecomment-1127032945 , I guess the same idea.

Reply:
@Yikun Good! I'll keep this idea for the next step.